persistence/persistence-mybatis/src/main/resources/org/cbioportal/persistence/mybatis/MutationMapper.xml

```diff
@@ -216,16 +218,18 @@
                 </trim>
                 )
             </if>
-            <if test="geneFilterQuery.getIncludeGermline() or geneFilterQuery.getIncludeSomatic() or geneFilterQuery.getIncludeUnknownStatus()">
+            <bind name="allMutationStatusSelected" value="geneFilterQuery.getIncludeGermline() and geneFilterQuery.getIncludeSomatic() and geneFilterQuery.getIncludeUnknownStatus()" />
+            <bind name="noMutationStatusSelected" value="not geneFilterQuery.getIncludeGermline() and not geneFilterQuery.getIncludeSomatic() and not geneFilterQuery.getIncludeUnknownStatus()" />
+            <if test="not allMutationStatusSelected and not noMutationStatusSelected">
                 <trim prefix="AND (" prefixOverrides="OR">
                     <if test="geneFilterQuery.getIncludeGermline()">
                         OR LOWER(mutation.MUTATION_STATUS) = 'germline'
                     </if>
                     <if test="geneFilterQuery.getIncludeSomatic()">
-                        OR LOWER(mutation.MUTATION_STATUS) = 'somatic'
+                        OR LOWER(mutation.MUTATION_STATUS) = 'somatic' 
                     </if>
                     <if test="geneFilterQuery.getIncludeUnknownStatus()">
-                        OR LOWER(mutation.MUTATION_STATUS) IN ('unknown', '')
+                        OR LOWER(mutation.MUTATION_STATUS) NOT IN ('somatic', 'germline')
```

Comment:
this logic should work for now but before enabling this on public portal we might want to play around with it to handle these cases:

![Screen Shot 2021-06-17 at 12 58 59 PM](https://user-images.githubusercontent.com/1334004/122579737-3796e200-d023-11eb-95fc-e7fe2dff5efc.png)

Another option might be to just leave mutation_status as is and not try to group them in somatic/germline/unknown. Then we would need to fix the data